bin/node/src/service.rs

```diff
@@ -252,16 +259,18 @@ fn setup(
             warp_sync: None,
         })?;
 
+    let chain_status = SubstrateChainStatus::new(backend.clone())
+        .map_err(|e| ServiceError::Other(format!("failed to set up chain status: {}", e)))?;
     let rpc_builder = {
         let client = client.clone();
         let pool = transaction_pool.clone();
-
         Box::new(move |deny_unsafe, _| {
-            let deps = crate::rpc::FullDeps::<Block, _, _> {
+            let deps: RpcFullDeps<Block, _, _, _> = crate::rpc::FullDeps {
                 client: client.clone(),
                 pool: pool.clone(),
                 deny_unsafe,
                 import_justification_tx: import_justification_tx.clone(),
+                justification_translator: chain_status.clone(),
```

Comment:
Yup:
```
267 |         Box::new(move |deny_unsafe, _| {
    |                  ^^^^^^^^^^^^^^^^^^^^^ this closure implements `FnOnce`, not `Fn`
...
273 |                 justification_translator: chain_status,
    |                                           ------------ closure is `FnOnce` because it moves the variable `chain_status` out of its environment
...
286 |         rpc_builder,
    |         ----------- the requirement to implement `Fn` derives from here
```

Interesting case though.